TACC/tup-ui
# Task/TUP-457/TUP-514 Add System Queue Details

## Overview
Created system monitor page that displays job queues and system monitor for each system

TODO:
Average wait time table added on and not ready from TAS yet.

<h3>Click the Jira task to see Design's mockups</h3>

## Related

- [TUP-457](https://jira.tacc.utexas.edu/browse/TUP-457)
-  [TUP-514](https://jira.tacc.utexas.edu/browse/TUP-514)

## Changes

- Created new system-status page
- Created hook for useSystemQueue to get queue details for each system
- Created system job queue table
- Created default prop for SystemDetails to select tas name from params and if none selected, choose the default system (Frontera for now)
- Created routes in App to get job queue for each system
- Changed system monitor table so that each system name links to the correct job queue table
- Created system monitor navigation bar
- Added navigation button for System status as per design team specs
- Added title to system monitor table

________New Changes for TUP-514________

- Used two column layout for the TODO average wait time table
- Default to Frontera when clicking on the system status nav icon
- Add system status table at top to show current selected system's status
- Only display System Status title for table on the Dashboard
- On system status page, add the selected system to the header of the page
- Changed column titles as per mockups

## Testing

1. Go to http://localhost:8000/portal/dashboard and click on either 

- a linked system name from the system monitor table, 
- or the new system status navigation button on the left.

2. This should bring you to a page with a system monitor table and a job queue table for that system. If you selected the System Status navigation icon, it should display Frontera by default and take you to the path /system-status. 

3. For Frontera, make sure that the stats match up with [TAS here](https://tap.tacc.utexas.edu/status/frontera/)

4. Check that the path looks like /system-status/{tas_name} if a system is selected from the second navigation on the system status page

## UI
Dashboard System Monitor

<img width="1502" alt="Screenshot 2023-07-12 at 9 53 34 AM" src="https://github.com/TACC/tup-ui/assets/96220951/d847f10d-5c8f-4e9f-84ca-06f96bd8ba87">

[/system_monitor ](http://localhost:8000/portal/system_monitor)- when clicking on System Status from the dashboard nav bar

<img width="1496" alt="Screenshot 2023-07-12 at 9 53 51 AM" src="https://github.com/TACC/tup-ui/assets/96220951/f91bc522-5bea-4691-b48d-1f73d15dd23a">


[/system_monitor/lonestar6](http://localhost:8000/portal/system_monitor/lonestar6)
<img width="1509" alt="Screenshot 2023-07-12 at 9 54 10 AM" src="https://github.com/TACC/tup-ui/assets/96220951/db9734e2-1f24-4e2f-99b9-2557550ef799">



## Notes

Next step will be to merge in [Wesley's content PR #257 ](https://github.com/TACC/tup-ui/pull/257) which will make this page into a two column grid layout to make space for average wait time table on the right. This will also add appropriate spacing between the system monitor row and the job queue table.
